Compiled from "Exception.scala"
public class scala.util.control.Exception$Catch<T> implements scala.util.control.Exception$Described {
  private final scala.PartialFunction<java.lang.Throwable, T> pf;

  private final scala.Option<scala.util.control.Exception$Finally> fin;

  private final scala.Function1<java.lang.Throwable, java.lang.Object> rethrow;

  private final java.lang.String name;

  private java.lang.String scala$util$control$Exception$Described$$_desc;

  public java.lang.String scala$util$control$Exception$Described$$_desc();
    Code:
       0: aload_0
       1: getfield      #23                 // Field scala$util$control$Exception$Described$$_desc:Ljava/lang/String;
       4: areturn

  public void scala$util$control$Exception$Described$$_desc_$eq(java.lang.String);
    Code:
       0: aload_0
       1: aload_1
       2: putfield      #23                 // Field scala$util$control$Exception$Described$$_desc:Ljava/lang/String;
       5: return

  public java.lang.String desc();
    Code:
       0: aload_0
       1: invokestatic  #35                 // Method scala/util/control/Exception$Described$class.desc:(Lscala/util/control/Exception$Described;)Ljava/lang/String;
       4: areturn

  public scala.util.control.Exception$Described withDesc(java.lang.String);
    Code:
       0: aload_0
       1: aload_1
       2: invokestatic  #40                 // Method scala/util/control/Exception$Described$class.withDesc:(Lscala/util/control/Exception$Described;Ljava/lang/String;)Lscala/util/control/Exception$Described;
       5: areturn

  public java.lang.String toString();
    Code:
       0: aload_0
       1: invokestatic  #44                 // Method scala/util/control/Exception$Described$class.toString:(Lscala/util/control/Exception$Described;)Ljava/lang/String;
       4: areturn

  public scala.PartialFunction<java.lang.Throwable, T> pf();
    Code:
       0: aload_0
       1: getfield      #47                 // Field pf:Lscala/PartialFunction;
       4: areturn

  public scala.Option<scala.util.control.Exception$Finally> fin();
    Code:
       0: aload_0
       1: getfield      #50                 // Field fin:Lscala/Option;
       4: areturn

  public scala.Function1<java.lang.Throwable, java.lang.Object> rethrow();
    Code:
       0: aload_0
       1: getfield      #53                 // Field rethrow:Lscala/Function1;
       4: areturn

  public java.lang.String name();
    Code:
       0: aload_0
       1: getfield      #55                 // Field name:Ljava/lang/String;
       4: areturn

  public <U> scala.util.control.Exception$Catch<U> or(scala.PartialFunction<java.lang.Throwable, U>);
    Code:
       0: new           #2                  // class scala/util/control/Exception$Catch
       3: dup
       4: aload_0
       5: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
       8: aload_1
       9: invokeinterface #65,  2           // InterfaceMethod scala/PartialFunction.orElse:(Lscala/PartialFunction;)Lscala/PartialFunction;
      14: aload_0
      15: invokevirtual #67                 // Method fin:()Lscala/Option;
      18: aload_0
      19: invokevirtual #69                 // Method rethrow:()Lscala/Function1;
      22: invokespecial #73                 // Method "<init>":(Lscala/PartialFunction;Lscala/Option;Lscala/Function1;)V
      25: areturn

  public <U> scala.util.control.Exception$Catch<U> or(scala.util.control.Exception$Catch<U>);
    Code:
       0: aload_0
       1: aload_1
       2: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
       5: invokevirtual #77                 // Method or:(Lscala/PartialFunction;)Lscala/util/control/Exception$Catch;
       8: areturn

  public <U> U apply(scala.Function0<U>);
    Code:
       0: aload_1
       1: invokeinterface #85,  1           // InterfaceMethod scala/Function0.apply:()Ljava/lang/Object;
       6: goto          60
       9: astore        5
      11: aload_0
      12: invokevirtual #69                 // Method rethrow:()Lscala/Function1;
      15: aload         5
      17: invokeinterface #90,  2           // InterfaceMethod scala/Function1.apply:(Ljava/lang/Object;)Ljava/lang/Object;
      22: invokestatic  #96                 // Method scala/runtime/BoxesRunTime.unboxToBoolean:(Ljava/lang/Object;)Z
      25: ifeq          35
      28: aload         5
      30: astore        9
      32: goto          112
      35: aload_0
      36: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
      39: aload         5
      41: invokeinterface #99,  2           // InterfaceMethod scala/PartialFunction.isDefinedAt:(Ljava/lang/Object;)Z
      46: ifeq          87
      49: aload_0
      50: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
      53: aload         5
      55: invokeinterface #100,  2          // InterfaceMethod scala/PartialFunction.apply:(Ljava/lang/Object;)Ljava/lang/Object;
      60: aload_0
      61: invokevirtual #67                 // Method fin:()Lscala/Option;
      64: dup
      65: astore_2
      66: invokevirtual #106                // Method scala/Option.isEmpty:()Z
      69: ifne          86
      72: aload_2
      73: invokevirtual #109                // Method scala/Option.get:()Ljava/lang/Object;
      76: checkcast     #111                // class scala/util/control/Exception$Finally
      79: invokevirtual #115                // Method scala/util/control/Exception$Finally.invoke:()V
      82: getstatic     #121                // Field scala/runtime/BoxedUnit.UNIT:Lscala/runtime/BoxedUnit;
      85: pop
      86: areturn
      87: aload         5
      89: astore        9
      91: goto          112
      94: astore        9
      96: aload_0
      97: invokevirtual #67                 // Method fin:()Lscala/Option;
     100: dup
     101: astore        6
     103: invokevirtual #106                // Method scala/Option.isEmpty:()Z
     106: ifne          140
     109: goto          125
     112: aload_0
     113: invokevirtual #67                 // Method fin:()Lscala/Option;
     116: dup
     117: astore        6
     119: invokevirtual #106                // Method scala/Option.isEmpty:()Z
     122: ifne          140
     125: aload         6
     127: invokevirtual #109                // Method scala/Option.get:()Ljava/lang/Object;
     130: checkcast     #111                // class scala/util/control/Exception$Finally
     133: invokevirtual #115                // Method scala/util/control/Exception$Finally.invoke:()V
     136: getstatic     #121                // Field scala/runtime/BoxedUnit.UNIT:Lscala/runtime/BoxedUnit;
     139: pop
     140: aload         9
     142: athrow
    Exception table:
       from    to  target type
           0     9     9   any
          87    94    94   any
           0    60    94   any

  public scala.util.control.Exception$Catch<T> andFinally(scala.Function0<scala.runtime.BoxedUnit>);
    Code:
       0: aload_0
       1: invokevirtual #67                 // Method fin:()Lscala/Option;
       4: astore_2
       5: getstatic     #133                // Field scala/None$.MODULE$:Lscala/None$;
       8: aload_2
       9: invokevirtual #136                // Method java/lang/Object.equals:(Ljava/lang/Object;)Z
      12: ifeq          49
      15: new           #2                  // class scala/util/control/Exception$Catch
      18: dup
      19: aload_0
      20: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
      23: new           #138                // class scala/Some
      26: dup
      27: new           #111                // class scala/util/control/Exception$Finally
      30: dup
      31: aload_1
      32: invokespecial #141                // Method scala/util/control/Exception$Finally."<init>":(Lscala/Function0;)V
      35: invokespecial #144                // Method scala/Some."<init>":(Ljava/lang/Object;)V
      38: aload_0
      39: invokevirtual #69                 // Method rethrow:()Lscala/Function1;
      42: invokespecial #73                 // Method "<init>":(Lscala/PartialFunction;Lscala/Option;Lscala/Function1;)V
      45: astore_3
      46: goto          97
      49: aload_2
      50: instanceof    #138                // class scala/Some
      53: ifeq          99
      56: aload_2
      57: checkcast     #138                // class scala/Some
      60: astore        4
      62: new           #2                  // class scala/util/control/Exception$Catch
      65: dup
      66: aload_0
      67: invokevirtual #59                 // Method pf:()Lscala/PartialFunction;
      70: new           #138                // class scala/Some
      73: dup
      74: aload         4
      76: invokevirtual #147                // Method scala/Some.x:()Ljava/lang/Object;
      79: checkcast     #111                // class scala/util/control/Exception$Finally
      82: aload_1
      83: invokevirtual #151                // Method scala/util/control/Exception$Finally.and:(Lscala/Function0;)Lscala/util/control/Exception$Finally;
      86: invokespecial #144                // Method scala/Some."<init>":(Ljava/lang/Object;)V
      89: aload_0
      90: invokevirtual #69                 // Method rethrow:()Lscala/Function1;
      93: invokespecial #73                 // Method "<init>":(Lscala/PartialFunction;Lscala/Option;Lscala/Function1;)V
      96: astore_3
      97: aload_3
      98: areturn
      99: new           #153                // class scala/MatchError
     102: dup
     103: aload_2
     104: invokespecial #154                // Method scala/MatchError."<init>":(Ljava/lang/Object;)V
     107: athrow

  public <U> scala.Option<U> opt(scala.Function0<U>);
    Code:
       0: aload_0
       1: invokevirtual #160                // Method toOption:()Lscala/util/control/Exception$Catch;
       4: new           #162                // class scala/util/control/Exception$Catch$$anonfun$opt$1
       7: dup
       8: aload_0
       9: aload_1
      10: invokespecial #165                // Method scala/util/control/Exception$Catch$$anonfun$opt$1."<init>":(Lscala/util/control/Exception$Catch;Lscala/Function0;)V
      13: invokevirtual #167                // Method apply:(Lscala/Function0;)Ljava/lang/Object;
      16: checkcast     #102                // class scala/Option
      19: areturn

  public <U> scala.util.Either<java.lang.Throwable, U> either(scala.Function0<U>);
    Code:
       0: aload_0
       1: invokevirtual #172                // Method toEither:()Lscala/util/control/Exception$Catch;
       4: new           #174                // class scala/util/control/Exception$Catch$$anonfun$either$1
       7: dup
       8: aload_0
       9: aload_1
      10: invokespecial #175                // Method scala/util/control/Exception$Catch$$anonfun$either$1."<init>":(Lscala/util/control/Exception$Catch;Lscala/Function0;)V
      13: invokevirtual #167                // Method apply:(Lscala/Function0;)Ljava/lang/Object;
      16: checkcast     #177                // class scala/util/Either
      19: areturn

  public <U> scala.util.Try<U> withTry(scala.Function0<U>);
    Code:
       0: aload_0
       1: invokevirtual #182                // Method toTry:()Lscala/util/control/Exception$Catch;
       4: new           #184                // class scala/util/control/Exception$Catch$$anonfun$withTry$1
       7: dup
       8: aload_0
       9: aload_1
      10: invokespecial #185                // Method scala/util/control/Exception$Catch$$anonfun$withTry$1."<init>":(Lscala/util/control/Exception$Catch;Lscala/Function0;)V
      13: invokevirtual #167                // Method apply:(Lscala/Function0;)Ljava/lang/Object;
      16: checkcast     #187                // class scala/util/Try
      19: areturn

  public <U> scala.util.control.Exception$Catch<U> withApply(scala.Function1<java.lang.Throwable, U>);
    Code:
       0: new           #191                // class scala/util/control/Exception$Catch$$anon$2
       3: dup
       4: aload_0
       5: aload_1
       6: invokespecial #194                // Method scala/util/control/Exception$Catch$$anon$2."<init>":(Lscala/util/control/Exception$Catch;Lscala/Function1;)V
       9: astore_2
      10: new           #2                  // class scala/util/control/Exception$Catch
      13: dup
      14: aload_2
      15: aload_0
      16: invokevirtual #67                 // Method fin:()Lscala/Option;
      19: aload_0
      20: invokevirtual #69                 // Method rethrow:()Lscala/Function1;
      23: invokespecial #73                 // Method "<init>":(Lscala/PartialFunction;Lscala/Option;Lscala/Function1;)V
      26: areturn

  public scala.util.control.Exception$Catch<scala.Option<T>> toOption();
    Code:
       0: aload_0
       1: new           #197                // class scala/util/control/Exception$Catch$$anonfun$toOption$1
       4: dup
       5: aload_0
       6: invokespecial #200                // Method scala/util/control/Exception$Catch$$anonfun$toOption$1."<init>":(Lscala/util/control/Exception$Catch;)V
       9: invokevirtual #202                // Method withApply:(Lscala/Function1;)Lscala/util/control/Exception$Catch;
      12: areturn

  public scala.util.control.Exception$Catch<scala.util.Either<java.lang.Throwable, T>> toEither();
    Code:
       0: aload_0
       1: new           #204                // class scala/util/control/Exception$Catch$$anonfun$toEither$1
       4: dup
       5: aload_0
       6: invokespecial #205                // Method scala/util/control/Exception$Catch$$anonfun$toEither$1."<init>":(Lscala/util/control/Exception$Catch;)V
       9: invokevirtual #202                // Method withApply:(Lscala/Function1;)Lscala/util/control/Exception$Catch;
      12: areturn

  public scala.util.control.Exception$Catch<scala.util.Try<T>> toTry();
    Code:
       0: aload_0
       1: new           #207                // class scala/util/control/Exception$Catch$$anonfun$toTry$1
       4: dup
       5: aload_0
       6: invokespecial #208                // Method scala/util/control/Exception$Catch$$anonfun$toTry$1."<init>":(Lscala/util/control/Exception$Catch;)V
       9: invokevirtual #202                // Method withApply:(Lscala/Function1;)Lscala/util/control/Exception$Catch;
      12: areturn

  public scala.util.control.Exception$Catch(scala.PartialFunction<java.lang.Throwable, T>, scala.Option<scala.util.control.Exception$Finally>, scala.Function1<java.lang.Throwable, java.lang.Object>);
    Code:
       0: aload_0
       1: aload_1
       2: putfield      #47                 // Field pf:Lscala/PartialFunction;
       5: aload_0
       6: aload_2
       7: putfield      #50                 // Field fin:Lscala/Option;
      10: aload_0
      11: aload_3
      12: putfield      #53                 // Field rethrow:Lscala/Function1;
      15: aload_0
      16: invokespecial #210                // Method java/lang/Object."<init>":()V
      19: aload_0
      20: invokestatic  #214                // Method scala/util/control/Exception$Described$class.$init$:(Lscala/util/control/Exception$Described;)V
      23: aload_0
      24: ldc           #216                // String Catch
      26: putfield      #55                 // Field name:Ljava/lang/String;
      29: return
}
